src/widgets.js

```diff
@@ -643,12 +645,31 @@ define([
         };
 
         widget.equivalentOption = function (val) {
-            val = val ? JSON.parse(val) : '';
+            function parseValue (val) {
+                try {
+                    return JSON.parse(val);
+                } catch(err) {
+                    return  val;
+                }
+            }
+
+            val = parseValue(val);
             return _.find(widget.getOptions(), function (option) {
-                return _.isEqual(option.value ? JSON.parse(option.value) : '', val);
+                return _.isEqual(parseValue(option.value), val);
             });
         };
 
+        widget.findOrAddCustom = function (value) {
```

Comment:
super picky, but `findOrAddCustom` sounds like it returns something, either a boolean or the custom option. `addCustomIfNeeded`?